docs/adding-new-resource.md

```diff
@@ -0,0 +1,705 @@
+testing. There are 3 steps we need to verify in manual tests: `Apply`, `Import`,
+`Delete`.
+
+**Apply:** We need to apply the example manifest to the cluster.
+
+
+```bash
+kubectl apply -f examples/redshift/v1beta1/endpointaccess.yaml
+```
+
+Successfully applying the example manifests to the cluster is only the first
+step. After successfully creating the Managed Resources, we need to check
+whether their statuses are ready or not. So we need to expect a `True` value for
+`Synced` and `Ready` conditions. To check the statuses of all created example
+manifests quickly you can run the `kubectl get managed` command. We will wait
+for all values to be `True` in this list:
+
+```bash
+NAME                            SYNCED   READY   EXTERNAL-NAME              AGE
+subnet.ec2.aws.upbound.io/bar   True     True    subnet-0149bf6c20720d596   26m
+subnet.ec2.aws.upbound.io/foo   True     True    subnet-02971ebb943f5bb6e   26m
+
+NAME                         SYNCED   READY   EXTERNAL-NAME           AGE
+vpc.ec2.aws.upbound.io/foo   True     True    vpc-0ee6157df1f5a116a   26m
+
+NAME                                                       SYNCED   READY   EXTERNAL-NAME              AGE
+cluster.redshift.aws.upbound.io/example-endpointaccess-c   True     True    example-endpointaccess-c   26m
+
+NAME                                                            SYNCED   READY   EXTERNAL-NAME            AGE
+endpointaccess.redshift.aws.upbound.io/example-endpointaccess   True     True    example-endpointaccess   26m
+
+NAME                                                            SYNCED   READY   EXTERNAL-NAME               AGE
+subnetgroup.redshift.aws.upbound.io/example-endpointaccess-sg   True     True    example-endpointaccess-sg   26m
+```
+
+As a second step, we need to check the `UpToDate` status condition. This status
+condition will be visible when you set the annotation: `upjet.upbound.io/test=true`.
+Without adding this annotation you cannot see the mentioned condition. The rough
+significance of this condition is to make sure that the resource does not remain
+in an update loop. To check the `UpToDate` condition for all MRs in the cluster,
+run:
+
+```bash
+kubectl annotate managed --all upjet.upbound.io/test=true --overwrite
+# check the conditions
+kubectl get endpointaccess.redshift.aws.upbound.io/example-endpointaccess -o yaml
+```
+
+You should see the output below:
+
+```yaml
+  conditions:
+  - lastTransitionTime: "2024-05-20T17:37:20Z"
+    reason: Available
+    status: "True"
+    type: Ready
+  - lastTransitionTime: "2024-05-20T17:37:11Z"
+    reason: ReconcileSuccess
+    status: "True"
+    type: Synced
+  - lastTransitionTime: "2024-05-20T17:37:15Z"
+    reason: Success
+    status: "True"
+    type: LastAsyncOperation
+  - lastTransitionTime: "2024-05-20T17:37:48Z"
+    reason: UpToDate
+    status: "True"
+    type: Test
+```
+
+When all of the fields are `True`, the `Apply` test was successfully completed!
+
+**Import**
```

Comment:
After multi-version CRD support, now we have a bit different import test mechanism. Please check the uptest implementation to understand the new Import steps. Still it is very common but we have some additional ones.

Reply:
We discussed it offline with @sergenyalcin  and decided that the current steps were sufficient since we added the v1beta1 version while adding a new resource. 

A new section has been added so that the user can use uptest locally.